src/EarnerManager.sol

```diff
@@ -0,0 +1,253 @@
+
+        EarnerDetails storage details_ = _earnerDetails[account_];
+
+        // NOTE: Not using `isInAdministratedEarnersList(account_)` here to avoid redundant storage reads.
+        return _isValidAdmin(details_.admin) ? (true, details_.feeRate, details_.admin) : (false, 0, address(0));
+    }
+
+    /// @inheritdoc IEarnerManager
+    function getEarnerDetails(
+        address[] calldata accounts_
+    ) external view returns (bool[] memory statuses_, uint16[] memory feeRates_, address[] memory admins_) {
+        statuses_ = new bool[](accounts_.length);
+        feeRates_ = new uint16[](accounts_.length);
+        admins_ = new address[](accounts_.length);
+
+        bool earnersListsIgnored_ = earnersListsIgnored();
+
+        for (uint256 index_; index_ < accounts_.length; ++index_) {
+            if (earnersListsIgnored_) {
+                statuses_[index_] = true;
+                continue;
+            }
+
+            address account_ = accounts_[index_];
+
+            if (isInRegistrarEarnersList(account_)) {
+                statuses_[index_] = true;
+                continue;
+            }
+
+            EarnerDetails storage details_ = _earnerDetails[account_];
+
+            // NOTE: Not using `isInAdministratedEarnersList(account_)` here to avoid redundant storage reads.
+            if (!_isValidAdmin(details_.admin)) continue;
+
+            statuses_[index_] = true;
+            feeRates_[index_] = details_.feeRate;
+            admins_[index_] = details_.admin;
+        }
+    }
+
+    /// @inheritdoc IEarnerManager
+    function isAdmin(address account_) public view returns (bool isAdmin_) {
+        // TODO: Consider transient storage for memoizing this check.
+        return IRegistrarLike(registrar).listContains(ADMINS_LIST_NAME, account_);
+    }
+
+    /* ============ Internal Interactive Functions ============ */
+
+    /**
+     * @dev   Sets the earner details for `account_`, assuming `msg.sender` is the calling admin.
+     * @param account_ The account under which yield could generate.
+     * @param status_  Whether the account is an earner, according to the admin.
+     * @param feeRate_ The fee rate to be taken from the yield.
+     */
+    function _setDetails(address account_, bool status_, uint16 feeRate_) internal {
+        if (account_ == address(0)) revert ZeroAccount();
+        if (!status_ && (feeRate_ != 0)) revert InvalidDetails(); // Fee rate must be zero if status is false.
+        if (feeRate_ > MAX_FEE_RATE) revert FeeRateTooHigh();
+        if (isInRegistrarEarnersList(account_)) revert AlreadyInRegistrarEarnersList(account_);
+
+        address currentAdmin_ = _earnerDetails[account_].admin;
+
+        // Revert if the details have already been set by an admin that is not `msg.sender`, and is still an admin.
+        if ((currentAdmin_ != address(0)) && (currentAdmin_ != msg.sender) && isAdmin(currentAdmin_)) {
```

Comment:
Can be simplified by using `_isValidAdmin()`:
```suggestion
        if (_isValidAdmin(currentAdmin_) && (currentAdmin_ != msg.sender)) {
```

Reply:
No,. because this way avoids potentially unnecessary contract call and storage reads if `currentAdmin_ == msg.sender`.